packages/marketplace/src/components/pages/developer-app-detail/app-detail/app-content/app-content.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react'
+import { GET_ALL_PAGE_SIZE } from '@/constants/paginator'
+import { appInstallationsRequestData } from '@/actions/app-installations'
+import { selectDeveloperId } from '@/selector'
+import developerAppDetailStyles from '@/styles/pages/developer-app-detail.scss?mod'
+import { FaExternalLinkAlt } from 'react-icons/fa'
+import { H5 } from '@reapit/elements'
+import appDetailStyles from '@/styles/blocks/app-detail.scss?mod'
+import ConfirmUninstall from '@/components/ui/app-installations/confirm-uninstall'
+import { PagedResultInstallationModel_, InstallationModel } from '@reapit/foundations-ts-definitions'
+import { handleUninstall, handleAfterClose } from '@/components/ui/app-installations/app-installations-modal'
+import AuthFlow from '@/constants/app-auth-flow'
+import '@/styles/vendor/slick.scss'
+import { useSelector, useDispatch } from 'react-redux'
+import { selectInstallationAppData } from '@/selector/installations'
+import { Table, Modal } from '@reapit/elements'
+import { generateColumns } from '@/components/ui/app-installations/installations'
+import clientAppDetailStyles from '@/styles/pages/client-app-detail.scss?mod'
+import { RenderWithHeader } from '../render-with-header'
+import { AppDetailDataNotNull } from '@/reducers/client/app-detail'
+import AppAuthenticationDetail from '@/components/ui/app-authentication-detail'
+
+export type AppContentProps = {
+  appDetailData: AppDetailDataNotNull
```

Comment:
This looks a bit confused actually because a property is required by default.

Reply:
It's an alias for 
`AppDetailModel & {
  apiKey?: string
}`

Because selector `selectAppDetailData` won't return null, but it's property will be partial by default